# Activity 1: Projectile Motion

In [79]:
import math
initial_velocity = float(raw_input('What is the initial velocity? '))
angle = float(raw_input('What is the angle in degrees? '))
print'Given an initial velocity of {:.1f} m/s and an angle of {:.1f} degrees' .format(initial_velocity, angle)
time_of_flight = (2 * initial_velocity * (math.sin(math.radians(angle)))) / 9.8
maximum_height_reached = ((initial_velocity ** 2) * ((math.sin(math.radians(angle)) ** 2))) / (2 * 9.8)
horizontal_range = ((initial_velocity ** 2) * ((math.sin(math.radians(2 * angle))))) / 9.80
print'The time of flight is {:.2f} s' .format(time_of_flight)
print'The maximum height reached is {:.2f} m' .format(maximum_height_reached)
print'The horizontal range is {:.2f} m' .format(horizontal_range)

What is the initial velocity? 20
What is the angle in degrees? 50
Given an initial velocity of 20.0 m/s and an angle of 50.0 degrees
The time of flight is 3.13 s
The maximum height reached is 11.98 m
The horizontal range is 40.20 m


### Questions

1) Did you use ints or floats to represent your data? Does it make a difference? Why?

I used floats to represent my data because floats include decimal points such as 3.13 whereas ints are whole numbers. So in order to fulfill the requirement of having two decimal points with each calculated answer, I needed to use floats. So, yes it does make a difference because of the fact that ints only will be whole numbers.


2) What happens if a user enters in non-numerical input or nothing at all? Why does this happen?

If the user enters non-numerical input or nothing at all a ValueError appears saying that string could not be converted to float. This is because float is only for numbers and non-numerical input or nothing at all is not a number so it cannot be converted into a number value which means that then the rest of the code which uses that float value to calculate the time of flight and other calculations, cannot be executed.

# Activity 2: Library Book Shelf

In [80]:
shelf_number = int(raw_input('What is the shelf number? '))
row          = int(shelf_number / 5)
unit         = int(shelf_number % 5)

print 'Shelf {} is unit {} in row {}' .format(shelf_number, unit, row)

What is the shelf number? 6
Shelf 6 is unit 1 in row 1


### Questions

1) Explain what the modulus operator computes and how you used it in your program.

The modulus operator computes the remainder after dividing one number by another. For example: 27 % 16 = 11 because 16 only goes into 27 once and the remainder is 11. In my program I used this operation to determine the remainder when the shelf number is divided by 5. This remainder is the unit that the shelf is in.


2) Copy your code into a new cell and then modify it so that it correctly identifies the shelf row and unit given a shelf number in the layout below:

In [72]:
shelf_number = int(raw_input('What is the shelf number? '))
row          = int(((shelf_number - 1) / 5) + 1)
unit         = int(((shelf_number - 1) % 5) + 1)

print 'Shelf {} is unit {} and row {}' .format(shelf_number, unit, row)

What is the shelf number? 6
Shelf 6 is unit 1 and row 2


# Activity 3: Shapes

In [73]:
from IPython.display import HTML, display
x      = int(raw_input('What is the x coordinate? '))
y      = int(raw_input('What is the y coordinate? '))
width  = int(raw_input('What is the width? '))
height = int(raw_input('What is the height? '))
color  = raw_input('What is the color? ')
html   = '''
<svg height="100" width="100">
  <rect x={} y={} width={} height={} fill={} />
</svg>
''' .format(x, y, width, height, color)
display(HTML(html))

What is the x coordinate? 0
What is the y coordinate? 0
What is the width? 75
What is the height? 75
What is the color? green


### Questions

1) Explain how you constructed the SVG code and then used it to show the shape.

I constructed the SVG (Scalable Vector Graphics) code by importing HTML and then enclosing the shape commands in a <svg> tag. This tag generated a canvas for the shape, and within the tag I was able to use .format to use the values I had previously asked the user for to create the shape.


2) Copy your code into a new cell and then modify it so that it draws a circle instead:

In [81]:
from IPython.display import HTML, display
cx            = int(raw_input('What is the circle x coordinate? '))
cy            = int(raw_input('What is the circle y coordinate? '))
radius        = int(raw_input('What is the radius? '))
outline       = raw_input('What is the outline color? ')
outline_width = int(raw_input('What is the outline thickness? '))
color         = raw_input('What is the color? ')
html   = '''
<svg height="100" width="100">
  <circle cx={} cy={} r={} stroke={} stroke-width={} fill={} />
</svg>
''' .format(cx, cy, radius, outline, outline_width, color)
display(HTML(html))

What is the circle x coordinate? 50
What is the circle y coordinate? 50
What is the radius? 40
What is the outline color? black
What is the outline thickness? 3
What is the color? red
